allennlp/nlp_api.py

```diff
@@ -0,0 +1,289 @@
+from typing import Callable, Dict, List, Union
+
+import torch
+
+from .common.checks import ConfigurationError
+from .common import Params
+
+
+class NlpApi:
```

Comment:
It's probably worthwhile to specify what the various assumed interfaces are for the embedders, recurrent layers, etc (in terms of assumed input and output shapes).